mypy/typeops.py

```diff
@@ -715,8 +715,8 @@ class Status(Enum):
             for name, symbol in typ.type.names.items():
                 if not isinstance(symbol.node, Var):
                     continue
-                # Skip "_order_" and "__order__", since Enum will remove it
-                if name in ("_order_", "__order__"):
+                # Skip "_ignore", "_order_" and "__order__", since Enum will remove it
+                if name in ("_ignore", "_order_", "__order__"):
```

Comment:
`("_ignore", "_order_", "__order__")` looks like a good candidate to be a constant 🤔

Reply:
I tried to add a constant (note that _ignore should be _ignore_ as @A5rocks observed below) to semanal_enum.py, but I got a circular import error:
```diff
diff --git a/mypy/semanal_enum.py b/mypy/semanal_enum.py
index e0f97df..bef8483 100644
--- a/mypy/semanal_enum.py
+++ b/mypy/semanal_enum.py
@@ -20,8 +20,9 @@
 ENUM_BASES: Final = frozenset((
     'enum.Enum', 'enum.IntEnum', 'enum.Flag', 'enum.IntFlag',
 ))
+ENUM_REMOVED_PROPS: Final = frozenset(('_ignore_', '_order_', '__order__'))
 ENUM_SPECIAL_PROPS: Final = frozenset((
-    'name', 'value', '_name_', '_value_', '_ignore_', '_order_', '__order__',
+    'name', 'value', '_name_', '_value_', *ENUM_REMOVED_PROPS,
     # Also attributes from `object`:
     '__module__', '__annotations__', '__doc__', '__slots__', '__dict__',
 ))
diff --git a/mypy/typeops.py b/mypy/typeops.py
index 201e76c..558191a 100644
--- a/mypy/typeops.py
+++ b/mypy/typeops.py
@@ -22,6 +22,7 @@
 )
 from mypy.maptype import map_instance_to_supertype
 from mypy.expandtype import expand_type_by_instance, expand_type
+from mypy.semanal_enum import ENUM_REMOVED_PROPS

 from mypy.typevars import fill_typevars

@@ -715,8 +716,8 @@ class Status(Enum):
             for name, symbol in typ.type.names.items():
                 if not isinstance(symbol.node, Var):
                     continue
-                # Skip "_ignore_", "_order_" and "__order__", since Enum will remove it
-                if name in ("_ignore_", "_order_", "__order__"):
+                # Skip these since Enum will remove it
+                if name in ENUM_REMOVED_PROPS:
                     continue
                 new_items.append(LiteralType(name, typ))
             # SymbolTables are really just dicts, and dicts are guaranteed to preserve
```
But then circular import error:
```
Traceback (most recent call last):
  File "/home/.../mypy/venv/bin/mypy", line 33, in <module>
    sys.exit(load_entry_point('mypy', 'console_scripts', 'mypy')())
  File "/home/.../mypy/venv/bin/mypy", line 25, in importlib_load_entry_point
    return next(matches).load()
  File "/usr/lib/python3.8/importlib/metadata.py", line 77, in load
    module = import_module(match.group('module'))
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 848, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/.../mypy/mypy/__main__.py", line 6, in <module>
    from mypy.main import main, process_options
  File "/home/.../mypy/mypy/main.py", line 13, in <module>
    from mypy import build
  File "/home/.../mypy/mypy/build.py", line 32, in <module>
    from mypy.semanal import SemanticAnalyzer
  File "/home/.../mypy/mypy/semanal.py", line 91, in <module>
    from mypy.messages import (
  File "/home/.../mypy/mypy/messages.py", line 37, in <module>
    from mypy.subtypes import (
  File "/home/.../mypy/mypy/subtypes.py", line 13, in <module>
    import mypy.applytype
  File "/home/.../mypy/mypy/applytype.py", line 4, in <module>
    import mypy.sametypes
  File "/home/.../mypy/mypy/sametypes.py", line 9, in <module>
    from mypy.typeops import tuple_fallback, make_simplified_union
  File "/home/.../mypy/mypy/typeops.py", line 25, in <module>
    from mypy.semanal_enum import ENUM_REMOVED_PROPS
  File "/home/.../mypy/mypy/semanal_enum.py", line 14, in <module>
    from mypy.semanal_shared import SemanticAnalyzerInterface
  File "/home/.../mypy/mypy/semanal_shared.py", line 18, in <module>
    from mypy import join
  File "/home/.../mypy/mypy/join.py", line 13, in <module>
    from mypy.subtypes import (
ImportError: cannot import name 'is_subtype' from partially initialized module 'mypy.subtypes' (most likely due to a circular import) (/home/.../mypy/mypy/subtypes.py)
```

Any ideas for a better place to put this constant?

Reply:
`types.py`?

Reply:
Added ENUM_REMOVED_PROPS to types.py, and imported that where needed.